hack/update-submodule.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# Update submodules HEAD to remote branches
+
+set -euo pipefail
+
+function help() {
+    echo "Usage: update_submodules.sh [OPTIONS]... branch_name..."
+    echo "Update submodules HEAD to remote branches"
+    echo "Must be ran in repo root directory"
+    echo ""
+    echo "Options:"
+    echo "-m   Specific modules to update, separated by spaces. Defaults to all submodules"
+    echo "-h   Shows usage text"
+    echo "Example:"
+    echo "update_submodules.sh master release-4.6"
+    echo "update_submodules.sh master -m \"kubernetes ovn-kubernetes\""
+}
+
+# Stages changes to the given submodule and makes a commit including those changes
+function generate_submodule_commit() {
+    local submodule=$1
+    cd $submodule
+    origin_url=$(git remote get-url origin)
+    short_head=$(git rev-parse --short HEAD)
+    long_head=$(git rev-parse HEAD)
+    cd ..
+    git add $submodule
+    # Commit changes if there are any
+    if git commit -m "[submodule][$submodule] Update to $short_head" -m "Bump to $origin_url/commit/$long_head" -m "This commit was generated using hack/update_submodules.sh"; then
+        echo "New commit for $submodule"
+    fi
+}
+
+# Creates a new branch based off the given branch with commits updating all submodules provided in a space separated list
+function update_submodules_for_branch() {
+    local base_branch=$1
+    local new_branch=$2
+    local modulelist=$3
+
+    if git branch -D $new_branch; then
+	echo "Deleted branch $new_branch"
+    fi
+    git checkout $base_branch -b $new_branch
+
+    # Generate a commit updating each submodule
+    for submodule in $modulelist; do
+        git submodule update --remote $submodule
```

Comment:
Did you confirm that `.gitmodules` gets updated if you are changing branches for the submodules?

Reply:
Yes i did

Reply:
I am not sure how that happens. Say the `kubernetes` submodule tracks `release-4.8` and I now want it to track `release-4.9`, how do I go about doing that?